### DISCUSSION TITLE
Free Surface Loads Extension

### DISCUSSION THREAD

**Discussion by MartinSalak** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#241] [Ideas]
Hello SAF community.
We would like to extend the object StructuralSurfaceActionFree. We are adding missing control over the direction of perpendicular axis of the load. We are also introducing validity which allows the free load to generate loads on multiple objects.

Existing object: **StructuralSurfaceActionFree**
Changed rows:
| Property | Example | Definition |
| --- | --- | --- |
| Coordinate system | Global<br>Local<br> **Member LCS** | Defines the direction of the loads generated on the surfaces <br>Global = loads will be in the global directions <br>Local = loads will be in the LCS of the load <br>Member LCS = loads will be in the LCS of the member they are being generated on |

New rows:
| Property | Example | Definition |
| --- | --- | --- |
| Validity | All,<br>minus Z,<br>minus Z zero,<br>Z zero,<br>plus Z,<br>plus Z zero,<br>From to | All = entities located above as well as below are subject to the defined load <br>Minus Z = All entities below <br>Minus Z zero = All entities below and in the plane of the load <br>Z zero = Only entities in the plane <br>Plus Z = All entities above <br>Plus Z zero = All the entities above and in the plane <br>From to = All entities in the defined range |
| Validity to [m] | 2.1 | Validity in the direction of positive Z axis of the load. Loads will be generated only below this distance. Required for Validity = From to |
| Validity from [m] | -3.4 | Validity in the direction of positive Z axis of the load. Load will be generated only above this distance. Required for Validity = From to |
| Local Z direction | Positive, Negative | Sets the direction of local Z axis of the load. Local Z axis is always perpendicular to the plane of the load. Positive means that the direction will be in the positive direction of global axis. For more information about the LCS of the load see notes |

Notes:
-	The LCS of the load is created based on the order of the nodes defining the load. X axis is in the direction from the first point to the second. Z axis is always perpendicular, and the direction is set by the property Local Z direction. Y axis is then set by the right-hand rule.
- Local Z direction = Positive means that Z direction will be in the direction of positive Z global direction, if it is not possible then in the positive direction of X axis, if that is also not possible then in the direction of positive Y global axis.
-	Above is in the direction of positive local Z axis from XY plane of the load or defined distance from that plane. Below is in the direction of negative local Z axis. In the example below load will be generated only between 3 and 4 m from the free load. 
![image](https://user-images.githubusercontent.com/89633938/206848503-e74da98e-97b1-4b8e-90e6-0c8c65dee402.png)
  
-	The load is projected always in the direction perpendicular to the load = in the direction of local Z axis.   
![image](https://user-images.githubusercontent.com/89633938/199990367-d5c6090b-8028-4338-8627-14a1f3db7c13.png)  

edits:
8.12.2022 - note explaining the positive direction of local Z axis
8.12.2022 - direction of Validity from changed to be the same as Validity to, example image updated

**Comment by egarciamendez** [comment]
Great addition to SAF! Is this also available on the current SCIA OpenAPI?

**Reply by MartinSalak** [reply]
SCIA API supports only partly the Free Surface Load. It is possible to assign it only in the global directions at the moment. But as this will be implemented to SAF, it will later work as described above for API as well.

**Comment by MartinSalak** [comment]
The proposal is published in the latest docs:
https://www.saf.guide/en/latest/loads/structuralsurfaceactionfree.html

**Comment by egarciamendez** [comment]
Any news regarding this feature?
